oauth-common/src/main/java/io/strimzi/kafka/oauth/services/OAuthMetrics.java

```diff
@@ -0,0 +1,179 @@
+        MetricConfig metricConfig = getMetricConfig();
+
+        return createMetrics(metricConfig, ctx, reporters);
+    }
+
+    private List<MetricsReporter> initReporters() {
+        AbstractConfig kafkaConfig = initKafkaConfig();
+        List<MetricsReporter> reporters = kafkaConfig.getConfiguredInstances(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
+                MetricsReporter.class);
+
+        JmxReporter reporter = new JmxReporter();
+        reporter.configure(configMap);
+
+        reporters.add(reporter);
+        return reporters;
+    }
+
+    private AbstractConfig initKafkaConfig() {
+        ConfigDef configDef = new ConfigDef()
+                .define(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
+                        ConfigDef.Type.LIST,
+                        Collections.emptyList(),
+                        new ConfigDef.NonNullValidator(),
+                        ConfigDef.Importance.LOW,
+                        CommonClientConfigs.METRIC_REPORTER_CLASSES_DOC);
+
+        return new AbstractConfig(configDef, configMap);
+    }
+
+    private KafkaMetricsContext createKafkaMetricsContext() {
+        HashMap<String, String> contextLabels = new HashMap<>();
+
+        // We don't know how to get clusterId here
+        //contextLabels.put("kafka.cluster.id", getClusterId());
+
+        // Broker side init code (for reference)
+        //if (config.usesSelfManagedQuorum) {
+        //    contextLabels.put("kafka.node.id", nodeId)
+        //} else {
+        //    contextLabels.put("kafka.broker.id", brokerId)
+        //}
+
+        String processRoles = config.getValue("process.roles");
+        if (processRoles == null || "[]".equals(processRoles)) {
+            String brokerId = config.getValue("broker.id");
+            if (brokerId != null) {
+                contextLabels.put("kafka.broker.id", brokerId);
+            }
+        } else {
+            String nodeId = config.getValue("node.id");
+            if (nodeId != null) {
+                contextLabels.put("kafka.node.id", nodeId);
+            }
+        }
+
+        if (config.getValue("client.id") != null) {
+            String clientId = config.getValue("client.id");
+            if (clientId != null) {
+                contextLabels.put("client.id", clientId);
+            }
+        }
+
+        addConfiguredContextLabels(contextLabels);
+        return new KafkaMetricsContext("strimzi.oauth", contextLabels);
+    }
+
+    private void addConfiguredContextLabels(HashMap<String, String> contextLabels) {
+        String prefix = "metrics.context.";
+        for (Map.Entry<String, ?> entry: configMap.entrySet()) {
+            if (entry.getKey().startsWith(prefix)) {
+                contextLabels.put(entry.getKey().substring(prefix.length()), String.valueOf(entry.getValue()));
+            }
+        }
+    }
+
+    private MetricConfig getMetricConfig() {
+
+        int numSamples = config.getValueAsInt("metrics.num.samples", 2);
```

Comment:
We could use the `CommonClientConfigs` constants here too

Reply:
Done.